File: frontend/src/dtos/models_dtos/project_dto.ts
import type { ProjectStatus } from "../enums/ProjectStatus"
import type { TechnologyDTO } from "./technology_dto"



export class ProjectDTO {
    id: number
    name: string
    status: ProjectStatus
    githubRepositoryUrl: string
    technologies: TechnologyDTO[]
    createdAt: Date
    

    constructor(id: number, name: string, gitHubRepositoryUrl: string ,status: ProjectStatus, technologies: TechnologyDTO[] , createdAt: Date) {
        this.id = id
        this.name = name
        this.status = status
        this.githubRepositoryUrl = gitHubRepositoryUrl
        this.technologies = technologies
        this.createdAt = createdAt
        
    }
}

File: backend/src/main/java/com/grouply/backend/project/project/Dtos/ProjectDTO.java
package com.grouply.backend.project.project.Dtos;
import com.grouply.backend.project.project.ProjectStatus;
import com.grouply.backend.technology.dto.TechnologyDTO;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProjectDTO {
    private Long id;
    private String name;
    private ProjectStatus status;
    private String githubRepositoryUrl;
    private LocalDateTime createdAt;
    private Set<TechnologyDTO> technologies;
}
